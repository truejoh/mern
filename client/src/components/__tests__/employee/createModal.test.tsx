import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';

import CreateModal from '../../employee/modals/createModal';

describe('Employee Create Modal', () => {
  it('renders component', () => {
    const props = {
      visible: true,
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(<CreateModal {...props} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders props', () => {
    const props = {
      visible: true,
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(
      <MockedProvider mocks={[]}>
        <CreateModal {...props} />
      </MockedProvider>,
    )
      .first()
      .shallow()
      .first()
      .shallow();

    expect(component.prop('visible')).toEqual(true);
    expect(component.prop('onOk')).toBeInstanceOf(Function);
    expect(component.prop('onCancel')).toBeInstanceOf(Function);
  });
});
