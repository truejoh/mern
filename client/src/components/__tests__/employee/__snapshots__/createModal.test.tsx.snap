// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Create Modal renders component 1`] = `
<Query
  displayName="Apollo(CreateModal)"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "banks",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "_id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "branchName",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 57,
        "start": 0,
      },
    }
  }
  skip={false}
  warnUnhandledError={true}
>
  <Component />
</Query>
`;
